### DISCUSSION TITLE
Issue with CUBASE 14 and xtouch+2 extenders

### DISCUSSION THREAD

**Discussion by limageurpublic** [bjoluc/cubase-mcu-midiremote#58] [Q&A]
Hi,
Hope you're fine and re thanks for your involvement and the new latest release of the script.

1.
So, I have an issue when closing and re starting CUBASE 14.0.10 ( windows 10 ) ...each time cubase lost the xtouch and the midi remote panel is empty.
Each time I re open cubase, I have to add a control surface in the midi panel and add the script, etc....
Do I make a mistake ? Is it a well known issue ? 

2.
Other issue, when closing cubase, not all the 3 devices ( 1 ext in USB, Xtouch in USB, and 1 ext in HTTP mode ) show "disconnected", the lcd still show old informations from some tracks....

Kind regards from France.

Screenshots and pictures related :

1 first openning and setup xtouch OK

![1 first openning and setup xtouch OK](https://github.com/user-attachments/assets/5cae354f-0554-4848-827d-fb88abbbb11a)

2 xtouchs when closing cubase

![2 xtouchs when closing cubase](https://github.com/user-attachments/assets/713a731f-4316-4799-848e-18bb67a06f79)

3 lost midiremote when re openning cubase

![3 lost midiremote when re openning cubase](https://github.com/user-attachments/assets/b743cb02-16e9-4296-a6a8-c65f3118a3b7)

**Comment by danimusicos** [comment]
problème réglé avec le 13.0.51 , et la 14.010

**Reply by limageurpublic** [reply]
Hello, Non, je suis en 14.0.10 et je dois "recréer" la surface à chaque lancement....Peut-être est-ce lié au fait que j'aie assemblé 2 extenders et une xtouch principale ?.... à suivre....

**Reply by danimusicos** [reply]
Faut bien modifier le script comme indiqué, et installer les fichiers au bonne endroit sur windows. Moi dès que je l'encre mon ne projet les extenders sont reconnus..

**Reply by limageurpublic** [reply]
Bonjour, c'est bien aussi de dire "bonjour" par pure politesse élémentaire...lol...
Cela fait assez longtemps que j'utilise ce script, tout est correctement installé et paramétré...
En outre, je ne comprends pas "...je l'encre mon ne projet...."
Bonne continuation à toi, je ne veux pas te déranger, j'attends plutôt un retour de Bjorn en direct comme par le passé.
Musicalement.

**Comment by danimusicos** [comment]
problème réglé avec le 13.0.51 , et la 14.010

**Reply by limageurpublic** [reply]
Hello, Non, je suis en 14.0.10 et je dois "recréer" la surface à chaque lancement....Peut-être est-ce lié au fait que j'aie assemblé 2 extenders et une xtouch principale ?.... à suivre....

**Reply by danimusicos** [reply]
salut, voici le comportement quand j'ouvre cubase 14.0.10 et ensuite un projet; Steinberg.

Eric Chabanne précisemment m'a confirmé que ce problème était réglé, en tout cas depuis la 14.010, chez moi çà roule. @+

![20241219_143523](https://github.com/user-attachments/asset

https://github.com/user-attachments/assets/72f8541f-a45b-4d8b-8fe2-28c0c6cdb76f

s/e1732fd6-a4d2-449c-be52-68a562009cf7)

**Comment by limageurpublic** [comment]
Pour votre information, le problème est "connu" et non encore résolu dans la version cubase 14.0.10....

Here is the link to the Cubase's forum : 
https://forums.steinberg.net/t/cubase-14-midi-remote-not-saved/953255/13
 
![Capture d’écran 2024-12-19 135521](https://github.com/user-attachments/assets/c8d09f44-b432-4f8e-9c5c-40b429fc5b38)

**Comment by limageurpublic** [comment]
Re, force est de constater, quand on lit le forum, que le comportement est
instable en fonction de l'utilisateur/la plateforme...mac et windows users
semblent avoir le même problème que moi. À moins que la procédure
d'installation du script 1.10 de Bjorn demande autre chose que de
simplement remplacer le script dans le dossier prévu à cet effet...
j'attends le retour de Bjorn qui doit savoir comment rectifier ce petit
souci.

Le jeu. 19 déc. 2024, 14:47, danimusicos ***@***.***> a
écrit :

**Reply by danimusicos** [reply]
re, moi j'ai juste modifié le script en fonction des extenders, je l'ai placé dans le répertoire préconisé, Par contre le fichier midiremote datait d'avril 2024 chez moi mais il change de date en fonction des mofif.
Bjorn ne répond quasiment plus depuis plusieurs mois, je crains qu'il soit surbooké.
[Behringer_X-Touch.zip](https://github.com/user-attachments/files/18198936/Behringer_X-Touch.zip)

**Reply by limageurpublic** [reply]
re,
Je ne sais pas à quoi sert ton fichier. merci quand même...lol...je ne suis pas très doué, sans doute !
j'ai voulu l’ouvrir avec wordpad mais la transcription est hs, les lignes de commandes sont impossible à lire et à modifier alors que le script "Behringer_xtouch.js" est parfaitement interprété et modifiable dans wordpad pour y indiquer les extenders et le main...

Habituellement, j'installe le script de Bjorn "behringer_xtouch.js" dans le dossier midi remote suivant :
![Capture d’écran 2024-12-19 154831](https://github.com/user-attachments/assets/40b68d90-bee7-46d7-8279-338727bfb628)

à suivre

**Comment by limageurpublic** [comment]
Hello,
C'est incompréhensible...preuve que la solution steinberg est instable...sans rien avoir changé, le script détecte à nouveau les surfaces de controle et les affichent correctement dans cubase à la réouverture du projet !
je n'en crois pas mes yeux de zicos fou.
c'est à n'y rien comprendre.
je te dirai si cela est stable ou si le problème "revient" à nouveau dans une future "ouverture" d'un projet cubase différent.
Bonne fin de journée.

ps : quelle musique fais tu ? où peut-on écouter ton travail ?

**Reply by danimusicos** [reply]
Eh oui, c'est ainsi. Super. Bravo. Travaille bien maintenant que tout est opérationnel. Midi remote n'est pas modifiable, seul le script 1.10 l'est. À plus

**Reply by limageurpublic** [reply]
Le script 1.5 de Bjorn était lui aussi modifiable; l'an passé...

**Comment by limageurpublic** [comment]
SOLUTION TROUVÉE, mais c'est idiot, il y a bien un bug dans cubase...

Si on "ferme" le projet ( fichier - fermer ), et que cubase revient à la fenêtre de sélection des projets ( voir capture d'écran ci-dessous ), et bien en ouvrant le projet, la surface de contrôle est reconnue et rappelée.
En revanche si on fait "fichier - quitter cubase", là, cubase ne rappellera pas la surface de contrôle à l'ouverture d'un projet !

cubase "vide" quelque chose à la fermeture "totale" et ne retrouve pas ses petits alors que si on maintien ouverte la fenêtre de sélection des projet en "fermant" seulement de projet, eh bien, cubase retrouve ses petits !
...
Nul !
![Capture d’écran 2024-12-19 160601](https://github.com/user-attachments/assets/665aa97c-4e41-4d9e-beef-a017e4b417cc)

**Reply by danimusicos** [reply]
Hello, j'avais exactement ce problème avec la version initiale de cubase. Avec des multiples essais j'ai pu stabiliser. Il faut faire remonter à steinberg, car on paye quand même des mises à jour qui ne sont pas censées casser les précédentes. D'autant que la version 13.0.41 fonctionne nickel, et problème est apparu avec la 13.0.50. J'ai signalé à Steinberg qui répond très très lentement. Eric CHabanne m'a dit qu'il l'avait signalé aux développeurs et que c'était réglé. Je leur ai même demandé d'inclure dans la version le scrip en demandant à BJORN. Mais comme je le dis, il doit être débordé car il n'y a rien de nouveau depuis juin. En tou cas, j'ai passé pas mal de temps à supprimer, enlever, et tout à coup ça a fonctionné. Depuis la version 14.0.10 c'est bon pour moi. voilà où se trouve le scipt K:\Documents\Steinberg\Cubase\MIDI Remote\Driver Scripts\Local\behringer\xtouch. K: c'est disque interne .

**Reply by limageurpublic** [reply]
Re; ça veut dire que tu n'utilises pas le script de Bjorn ? tu mets ton script ( celui que tu m'as envoyé ) au  même endroit que celui où je mets celui de bjorn... je comprends pas tout. à suivre !

**Reply by danimusicos** [reply]
bonjour, j'ai voulu tester le fonctionnement sur mon système. Si utilise le tout dernier que j'avais téléchargé sur ce groupe courant juin. J'ai donc désinstallé le script quitté cubase .
Après redémarrage, J'ai importé le script que je t'ai envoyé, reparamétré mes deux machines. J'ai sauvegardé mon projet, quitté cubase. Relancé, le projet et les 2 x-touch ont fonctionné normalement.  Ce qui semble démontrer que ce n'est probablement pas cubase mais l'emplacement du script et sa version. Mais il est important d'utilise le midiremote que j'ai envoyé. De plus tous les boutons s'allument : flip, cycle, crub. @+

**Reply by limageurpublic** [reply]
Hello, je ne comprends pas . Tu dis que c'est sans doute lié à l'endroit où est posé le script...mais je pose le script de Bjorn exactement au même endroit que toi. Je ne comprends donc pas la solution que tu indiques. De plus, ton script que tu m'as envoyé est un fichier ".json" mais pas le script de Bjorn. Je ne sais pas où tu as trouvé le script que tu m'as envoyé. À suivre.

**Reply by danimusicos** [reply]
Salut, j'ai téléchargé le script sur ce groupe et pas ailleurs. Il s'agit du 10.0.1 de mémoire. Quant au midi remote c'était Bjorn qui me l'avait adressé il y a un certain temps.

**Reply by limageurpublic** [reply]
Hello, je ne comprends pas . 

tu me parles d'un script et d'un "midi remote"....cela veut dire que tu parles de deux scripts différents posés à différents endroits ?

le script de Bjorn s'appelle "behringer_xtouch.js" et se dépose dans le répertoire que tu indiques "\Documents\Steinberg\Cubase\MIDI Remote\Driver Scripts\Local\behringer\xtouch"

Donc où déposes-tu ce second script "Behringer_X-Touch.midiremote" que tu m'as envoyé ???? et à quoi sert-il si on a déjà posé le script de Bjorn ?

je regrette mais ce n'est pas clair. je suis sans doute idiot mais j'aimerais comprendre parce que, en outre, la solution de Kolutshan n'a pas fonctionné...j'ai intégré sa modif dans le script de Bjorn "behringer_xtouch.js" version v1.10.1 du 2 juin 2024 mais quand je quitte totalement cubase 14.0.10, les xtouch ne sont pas reconnus à la réouverture de cubase et la section midiremote est "vide" et je dois recommencer le paramétrage ...

à suivre.

**Comment by limageurpublic** [comment]
Hi everybody, "IN ENGLISH"
I have the same issue than many of you.
When re opening cubade, the xtouch/midi remote controlers disappeared…but :
At home, windows 10/cubase 14.0.10, if I completely QUIT cubase, the issue is here and when reopen cubase, I have to setup newly my midi remote…
If I only CLOSE cubase project and cubase let the selection projects windows opened. When I select a project in the list, at opening, all devices on midi remote are properly recalled .!!!

There is something that cubase dump some temp files ( I could imagine ) related to midi remote when completely closed, but cubase keep some temp files if you just shut down “project” .
Hope Steinberg could listen it quick, in fact when you close computer for upgrade or any other case. Cubase lost midi remote controlers set up before!
Kind regards from France.

**Comment by kolutshan** [comment]
Hi there and sorry for english only as I don't speak french. I had the same problem but asked someone for help and he actually found the solution! Here is what I had to change in the already imported "behringer_xtouch.js" script:

1. quit Cubase
2. open the folder where Cubase stores the Behringer X-Touch midi remote script
3. **make a copy of "behringer_xtouch.js"**
4. open "behringer_xtouch.js" in an editor
5. search for "X-Touch-Ext"
6. you should find the line: `var name = "X-Touch-Ext";` (it is line 4529 on my end)
7. replace "X-Touch-Ext" with: "X-Touch-Ext X-TOUCH_INT" the line should now look like this: `var name = "X-Touch-Ext X-TOUCH_INT";`
8. Reopen Cubase and it might already detect your controller(s) and if not remove the current connection and add it again

I hope that works out for you and thanks for this awesome script!

**Reply by limageurpublic** [reply]
Hi
Apologize for my bad skills in english.
many thanks for your answer but...when changing the script with your suggestion, it doesn't work.
i could imagine I made a mistake either installing wrong the script or with the script/cubase it self.
I am so disapointed.
Awaiting for help further more..
kind regards from France.

**Reply by kolutshan** [reply]
Hi, can you post a screenshot of the midi remote setup (input & output ports) with your X-Touch controllers in Cubase? Maybe the device ports have different names on your system and therefor don't match the configuration. As an example here is a picture of my setup:
![image](https://github.com/user-attachments/assets/9b96942f-cf9f-4ec7-8ed9-dd78d2ca34f3)

**Reply by limageurpublic** [reply]
Hey,
thanks for your help....it's becoming weird !
Look at the picture and let me know if there's something wrong .
As I said, if a don't quit totally cubase and let the "cubase project window" only closing previous project, cubase recall everything good, but by quitting totally cubase, at the re openning, the midi remote is empty and I have to set up anew my xtouchs devices as shown.
It drives me crazy !
Kind regards.

![Capture d’écran 2024-12-22 155820](https://github.com/user-attachments/assets/1fb1140a-af4c-468b-9de5-88fe63db0723)

**Reply by kolutshan** [reply]
Hi, ok it seems you have 1x "X-Touch extender" on the left, 1x "X-Touch" in the center and 1x "X-Touch Extender" on the right, correct?
And I'm not sure why there is a totally different device assigned on "Port d'entrée #3"?
However it seems that your main "X-Touch" controller has a different name than mine so maybe try changing the line to this:
`var name = "X-Touch-Ext X-TOUCH";`

**Reply by limageurpublic** [reply]
Hi,

1st : I have Two extenders and one xtouch main, but cubase doesn't accept 2 makies unit with the same name "x-touch exender', so i have to put the second extender in HTTP mode whose name is DESKTOP.-7xxxx

2nd : the script with your new suggestion and it doesn't work ! The "x touch/extenders" are not properly recalled, the midi remote panel is empty. I have to re setup anew !

At each "re setup", everything is OK until I close completely CUBASE prgm.

Grrrrr.....

**Reply by kolutshan** [reply]
Hi,
I see, the fact that you have a third device name (for the IP based X-Touch controller) is the issue here regarding the changes I've made to the script. Lets try what happens if you change the line to this (I can only do trial and error here):
`var name = "X-Touch-Ext X-Touch DESKTOP-7V9FN99";`
Unfortunately I can't forsee what will happen and therefore you should have a backup of your original and working (at least until Cubase is quit) midi remote script

**Comment by limageurpublic** [comment]
Hey,

thanks.for your involvement. you're so kind.

while i was awaiting for your answer I did the same modification as you mentionned last.
but it's still the same behaviour.

In fact, "the midi remote panel" in cubase is empty and the 3 devices ( at project's openning ) awakenned and show all the tracks without "color lcd" ( only standar color like it was not connected ).....all tracks visibles but the "midi remote" is still messing!

I could imagine it lefts a "bit" something that could make it work well but I don't know what it could be ? I am not a geek coder, I am just a musician trying to make music , lol.

![Capture d’écran 2024-12-22 172455](https://github.com/user-attachments/assets/0e9c242f-159e-4f73-80d4-6af88c5adc9e)
![1734884526964](https://github.com/user-attachments/assets/ac40fd93-7404-4767-a285-ac69a1b7d1ee)
![1734884601406](https://github.com/user-attachments/assets/5ebf0350-58a8-48ce-9bd3-6e15303e1869)

**Reply by limageurpublic** [reply]
is there anything related about 2 extenders config ???? as the script mentionned ( see the picture )
![Capture d’écran 2024-12-22 173011](https://github.com/user-attachments/assets/a537f28e-7b4f-48f0-9eeb-91b2c38d3383)

**Reply by kolutshan** [reply]
I would not change anything else in the script and I don't believe its the reason why it is not yet working for you. 
What happens if you add the X-Touch remote again and leaving the script as it is now?

**Reply by limageurpublic** [reply]
At re openning cubase, the midi remote is empty, the 3 devices awaken with track names but no color and no shortcutS/functions working...until I setup in midi remote and everything going back to "OK"....

I tried to go back to "1.5 version of Bjorn script" and open a project with "cubase 13"....and everything works fine, midi remote is showing the 3 devices and recall  perfectly all tracks on 3 devices...etc...the fact is there's something changed in cubase 14 that "corrupt" the script...

**Comment by limageurpublic** [comment]
Re,
I had to go back to cubase 13.0.10 due to this steinberg issue. Does anybody know if cubase's developpers have solve the problem about "2 mackie device with the same name" ? 

As I said, Berhinger did not put a menu to set the "internal" name of the device that it will send to cubase, in a his firmware ( as we can do for MDI/USB/HT mode, LCD brightness, etc, on start up), 

I've try to put my 2 extenders and my xtouch in USB mode but the second extender is not recognized by cubase because it share the same "device name" than with first extender....so, perhaps I have misunderstaood but is seems that REAPER accept 2 extenders in USB/mackie mode fully...not CUBASE !

Let's be patient.
